src/components/application_manager/src/hmi_state.cc

```diff
@@ -305,7 +319,9 @@ std::ostream& operator<<(std::ostream& os, const HmiState::StateID src) {
 }
 
 std::ostream& operator<<(std::ostream& os, const HmiState& src) {
-  os << "HMIState(app id:" << src.app_->app_id() << ", state:" << src.state_id()
+  const ApplicationSharedPtr app =
+      src.app_mngr_.application_by_hmi_app(src.hmi_app_id_);
+  os << "HMIState(app id:" << app->app_id() << ", state:" << src.state_id()
```

Comment:
@shoamano83 I would like to print out `hmi_app_id` as well as after your updates it will be widely used inside `HmiState`

Reply:
Sure, updated with commit https://github.com/smartdevicelink/sdl_core/pull/2225/commits/2f9108b62aed4679768cd2fff1b04887990bc47a.